core/executerbechmark_test.go

```diff
@@ -0,0 +1,445 @@
+package core_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/dicedb/dice/core"
+	"github.com/xwb1989/sqlparser"
+)
+
+var itr int
+
+type tupple struct {
+	key   string
+	value string
+}
+
+var keys = []struct {
+	count int
+}{
+	{count: 100},
+	{count: 1000},
+	{count: 10000},
+	{count: 100000},
+}
+
+func populateData(count int) {
+	if itr == 0 {
+		dataset := []tupple{}
+
+		for i := 0; i < count; i++ {
+			dataset = append(dataset, tupple{fmt.Sprintf("k%d", i), fmt.Sprintf("v%d", i)})
+		}
+
+		// Delete all keys
+		for _, data := range dataset {
+			obj := core.Get(data.key)
+			if obj != nil {
+				core.Del(data.key)
+			}
+		}
+
+		// Insert all keys
+		for _, data := range dataset {
+			core.Put(data.key, &core.Obj{Value: data.value})
+		}
+		itr++ // Set count to 1 to indicate data has been populated (Prevent Benchamarker to again populate the data)
+	}
+}
+
+func BenchmarkExecuteQueryOrderBykey(b *testing.B) {
+	for _, v := range keys {
+		itr = 0
+		populateData(v.count)
+
+		query := core.DSQLQuery{
+			KeyRegex: "k*",
+			Selection: core.QuerySelection{
+				KeySelection:   true,
+				ValueSelection: true,
+			},
+			OrderBy: core.QueryOrder{
+				OrderBy: "$key",
+				Order:   "asc",
+			},
+		}
+		// Reset the timer to exclude the setup time from the benchmark
+		b.ResetTimer()
+
+		b.Run(fmt.Sprintf("keys_%d", v.count), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				//nolint: errcheck
+				core.ExecuteQuery(query) // no need to check fo return values that;s covered in tests
```

Comment:
We should not ignore the return value even for the benchmark. While there is no need to check the query result, we should validate that we are not getting an error result (what if the executor bugs out and immediately returns an error, our benchmark would be left assuming that the query ran absurdly fast).

Reply:
Implemented error check for the `core.ExecuteQuery` function in the benchmarking